packages/analytics-js/src/components/configManager/util/validate.ts

```diff
@@ -41,19 +41,19 @@ const getTopDomainUrl = (url: string) => {
   // Create a URL object
   const urlObj = new URL(url);
 
-  // Extract the hostname and protocol
-  const { hostname, protocol } = urlObj;
+  // Extract the host and protocol
+  const { host, protocol } = urlObj;
```

Comment:
Any specific reason to change hostname to host?
For this URL: http://example.org:8888/foo/bar#bang
hostname will be `example.org`
host will be `example.org:8888`

Reply:
Yes, that's the reason. Since we've to request the backend server, `host` will be more appropriate than `hostname` with full port information.
Host name is appropriate for the TLD for the cookies.